### PR TITLE
Cloud Main Screen

### LINKED ISSUE #20: Introduce a Cloud Screen
Create a Cloud Screen as per described in the app flow chart.  

- App Flow - [Link](https://drive.google.com/file/d/1uBLJ7LRJJiDNLdo9pVEqlX6PpncQcvyD/view?usp=sharing)

_Note : Make the screen in such a way so that we can integrate (fetch its content) with backend in future. In simpler words use demo data (json) and parse it in runtime to mock the behaviour of a network call, you would need to use FutureBuilder for awaitng the load of data._

### PR DESCRIPTION
**Pull request by ItsAdityaKSingh**
Fixes #20

## Introduced a Cloud main page displaying different genres and resources available.

> Dummy Json is being used to fetch data right now. Upon API endpoint finalisation other functionalities can be improved.

### CODE REVIEW

File: lib/widgets/gridcards.dart [hunk 1]
```diff
@@ -19,11 +20,15 @@ class HomeScreenGrid extends StatelessWidget {
...
-                image: CloudIcon,
-                title: "Cloud",
-                // route: PeoplePage(),
-              ),
+              GestureDetector(
+                  onTap: () {
+                    Navigator.push(
+                      context,
+                      MaterialPageRoute(
+                          builder: (context) => const CloudPage()),
```
Comment (Arjupta) [inline lib/widgets/gridcards.dart]: Use Navigator.pushNamed after registering a namedRoute in the main.dart.

File: lib/widgets/cloud_carousel.dart [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    .toList();
+
+class CloudCarousel extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      child: Column(
+        children: [
+          CarouselSlider(
+            options: CarouselOptions(
```
Comment (Arjupta) [inline lib/widgets/cloud_carousel.dart]: Why this is a Column

File: lib/utils/constants.dart [hunk 3]
```diff
@@ -2,6 +2,7 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 
 // Images ->
+const String Back = "assets/back.png";
 const String SochemIcon = "assets/sochem.jpeg";
 const String ExampleProfile = "assets/sampleprofile.jpeg";
```
Comment (Arjupta) [inline lib/utils/constants.dart]: BackIcon will be a better name

File: lib/screen/cloud.dart [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+              ),
+            ],
+          ),
+          Container(
+            margin: const EdgeInsets.only(top: 10.0),
+            height: screensize.height * 0.635,
+            child: ListView(
+              shrinkWrap: true,
+              scrollDirection: Axis.vertical,
+              children: [
```
Comment (Arjupta) [inline lib/screen/cloud.dart]: All the children of this ListView should also be generated the same way as cloud_carousel.dart

**Review by Arjupta — CHANGES_REQUESTED** [review]
@ItsAdityaKSingh take a look

File: lib/widgets/cloud_carousel.dart (hunk 2 above)
Comment (ItsAdityaKSingh) [reply lib/widgets/cloud_carousel.dart]: My bad! I tried to add a slide-marker below but canceled the idea as it seemed useless here. Gonna remove it!

File: lib/widgets/gridcards.dart (hunk 1 above)
Comment (ItsAdityaKSingh) [reply lib/widgets/gridcards.dart]: This was added temporarily here. I guessed we could implement this once all the pages have been structured and functioning. Should I do so before?

File: lib/utils/constants.dart (hunk 3 above)
Comment (ItsAdityaKSingh) [reply lib/utils/constants.dart]: Should we instead of adding an additional icon, just use some custom icon provided by Flutter? What's your view?

Comment (Arjupta) [reply lib/utils/constants.dart]: Yeah that is preffered

**Review by Arjupta — COMMENTED** [review]
@ItsAdityaKSingh everything looks fine. Just resolve the conflicts and the changes left (check comments).

**Review by Arjupta — APPROVED** [review]
LGTM

**Pull request merged**